internal/pb/fs.proto

```diff
@@ -70,6 +72,7 @@ message ObjectQuery {
     bool is_prefix = 2;
     bool with_content = 3;
     repeated string ignores = 4;
+    bool with_hash = 5;
```

Comment:
I don't think we should add `with_hash` here.

This is the public facing API for DateiLager and `with_hash` is only ever used internally. Instead we should just make this a function argument to the query builder and not something we expose.